applications/StructuralMechanicsApplication/tests/cpp_tests/test_metric_process.cpp

```diff
@@ -0,0 +1,351 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "geometries/triangle_2d_3.h"
+#include "geometries/tetrahedra_3d_4.h"
+#include "testing/testing.h"
+#include "includes/kratos_flags.h"
+#include "includes/gid_io.h"
+
+/* Processes */
+#include "processes/compute_nodal_gradient_process.h"
+#include "custom_processes/metrics_spr_error_process.h"
+
+namespace Kratos 
+{
+    namespace Testing 
+    {
+        typedef Node<3> NodeType;
+      
+        void GiDIODebugMetricSPR(ModelPart& ThisModelPart)
+        {
+            GidIO<> gid_io("TEST_METRIC_SPR", GiD_PostBinary, SingleFile, WriteUndeformed,  WriteElementsOnly);
```

Comment:
the test should check separately the ZZ estimator and the metric calculation, since they should be independent features

Reply:
Ok